File: backend/src/main/java/pl/migibud/yourDoctor/appointment/Appointment.java
package pl.migibud.yourDoctor.appointment;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import pl.migibud.yourDoctor.address.Address;
import pl.migibud.yourDoctor.doctor.Doctor;
import pl.migibud.yourDoctor.user.AppUser;

import javax.persistence.*;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;

@Entity
@Table(name = "appointment")
@Getter
@Setter(value = AccessLevel.PACKAGE)
@NoArgsConstructor
public class Appointment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private LocalDate dateOfAppointment;
    private LocalTime timeOfAppointment;
    private LocalDateTime registrationTime;
    @CreationTimestamp
    private LocalDateTime createdAt;
    @UpdateTimestamp
    private LocalDateTime updatedAt;
    private boolean taken;
    @OneToOne
    @JoinColumn(name = "address_id")
    @JsonIgnore
    private Address address;

    @ManyToOne
    @JoinColumn(name = "doctor_id")
    @JsonIgnore
    private Doctor doctor;

    @ManyToOne
    @JoinColumn(name = "user_id")
    @JsonIgnore
    private AppUser appUser;

    Appointment(LocalDate dateOfAppointment, LocalTime timeOfAppointment, Address address, Doctor doctor) {
        this.dateOfAppointment = dateOfAppointment;
        this.timeOfAppointment = timeOfAppointment;
        this.address = address;
        this.doctor = doctor;
    }
}

File: frontend/src/app/models/appointment.ts
import { AddressDto } from "./address"
import { DoctorForUserDto } from "./doctor"
import { AppUserDto } from "./user"

export type AppointmentDto = {
  id: number | null,
  dateOfAppointment: string,
  timeOfAppointment: string,
  doctor: DoctorForUserDto,
  address: AddressDto,
  taken: boolean,
}

export type CreateAppointmentRequest = {
  doctorId:number,
  addressId:number,
  dateOfAppointment: string,
  timeOfAppointment: string,
}

export type CreateAppointmentRegistrationRequest = {
  appointmentId:number,
  userId:number|null,
}

export type AppointmentForUserDto = {
  id: number | null,
  dateOfAppointment: string,
  timeOfAppointment: string,
  doctor: DoctorForUserDto,
  address: AddressDto,
  taken: boolean,
}

export type AppointmentForDoctorDto = {
  id: number | null,
  dateOfAppointment: string,
  timeOfAppointment: string,
  user: AppUserDto,
  address: AddressDto,
  taken: boolean,
}
